### PR TITLE
docs: add proposal how users should indicate application protocol of a service

### PR DESCRIPTION
**Pull request by yskopets**
### Summary

* add proposal how users should indicate application protocol of a service

### CODE REVIEW

File: docs/proposals/L7-protocols.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+* however, they are not a part of this proposal
+* we could add support for it in the future, probably in the form of recommendations
+
+### About "protocol" tag in case of ServiceProfile
+
+* If it is important to have `protocol` tag for the sake of `Kuma` policy matching (e.g., to be able to apply a policy to all `http` services), this tag could be generated automatically out of a `ServiceProfile`
+
+## Open questions
+
+1. What happens if a user defines different `protocol` tag for the same `service` in different `Dataplanes` ?
```
Comment (jakubdyszkiewicz) [inline docs/proposals/L7-protocols.md]: We need this information when generating inbound listeners, right? We generate listener for a given dataplane, so if one DP is defined with TCP then TCP listener will be generated, if DP is defined with HTTP then HTTP listener will be generated.

File: docs/proposals/L7-protocols.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+* REST API
+  ```yaml
+  type: ServiceProfile
+  mesh: default
+  name: backend # the same value as in `service` tag
+  profiles:
+  - api:
+      rest:
+        spec:
+          url: https://generator.swagger.io/api/swagger.json
```
Comment (jakubdyszkiewicz) [inline docs/proposals/L7-protocols.md]: What can we do with such information?
LinkerD has this to generate endpoints for which then metrics are collected, but we don't need it once generate listener with http connection manager filter.

File: docs/proposals/L7-protocols.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    name: backend
+    namespace: example
+  spec:
+    ports:
+    - port: 8080
+      name: http          # `http` protocol in the `name`
+    - port: 7070
+      name: http-metrics  # `http` protocol in the `name`
+    - port: 6060
+      name: somethinggrpc # `grpc` protocol in the `name`
```
Comment (jakubdyszkiewicz) [inline docs/proposals/L7-protocols.md]: What if I have both in the name? Example: grpc-http-bridge

**Comment by subnetmarco** [comment]
For now just add `protocol`. I am sure that as we add more features there will be a greater need for something similar to `ServiceProfile`, but we don't have to make this call now.

**Pull request merged**